### DISCUSSION TITLE
Trivy not identifying fixed package version in Debian

### DISCUSSION THREAD

**Discussion by neo-travis-friesen** [aquasecurity/trivy#7014] [Q&A]
### Question

The short story is we're building some docker images based off of debian stable, aka `bookworm`. Trivy correctly identifies that some of the installed packages have vulnerabilities. Fixes are not yet available for stable, so our solution was to modify the debian.sources and use the `testing` (aka trixie, I think) distribution to install versions of the packages that do have fixes.

While we can verify that the new version of the package is successfully installed, and Trivy can see that the installed package version is updated, it still reports the package as vulnerable even though the installed version exceeds the vulnerable version. 

For example, the sudo package, Trivy can clearly see that the installed package is >1.9.15, but it still reports the vulnerability despite it being fixed in 1.9.15:

```
        {
          "VulnerabilityID": "CVE-2023-42465",
          "PkgID": "sudo@1.9.15p5-3+b1",
          "PkgName": "sudo",
          "PkgIdentifier": {
            "PURL": "pkg:deb/debian/sudo@1.9.15p5-3%2Bb1?arch=amd64&distro=debian-12.5",
            "UID": "f70400c68d19fa66"
          },
          "InstalledVersion": "1.9.15p5-3+b1",
          "Status": "affected",
          "Layer": {
            "Digest": "sha256:eb28f8da5305eb65670de1f885788779d1301b13935a2f975a8552b8f77ff151",
            "DiffID": "sha256:8240dc20caccd95e83cd39c4da845e2209852c6871d38f5204da7fe9cfdb8539"
          },
          "SeveritySource": "nvd",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2023-42465",
          "DataSource": {
            "ID": "debian",
            "Name": "Debian Security Tracker",
            "URL": "https://salsa.debian.org/security-tracker-team/security-tracker"
          },
          "Title": "sudo: Targeted Corruption of Register and Stack Variables",
          "Description": "Sudo before 1.9.15 might allow row hammer attacks (for authentication bypass or privilege escalation) because application logic sometimes is based on not equaling an error value (instead of equaling a success value), and because the values do not resist flips of a single bit.",
```

This seems to be a manifestation of this similar issue: https://github.com/aquasecurity/trivy/discussions/5405

My guess is what's happening is that Trivy is relying on the security reports for our release, bookworm. But since there is no fixed version in bookworm, the security database (the exact term slips my mind right now) doesn't list a fixed package version; the fact that we are now on a fixed version from testing is irrelevant, as bookworm's security database doesn't know that this is a fixed version.

My question is about how best to resolve this issue. Modules and rego policy files may be one possibility, but I'd hate to be writing exceptions for every single package and vulnerability. Seems like a bit of a maintenance nightmare. What I'd like to be able to do is augment the security database with information about fixed versions, or to be able to selectively apply the security database from a newer release to the one that trivy is using for its automatically detected OS version.

Thanks in Advance.

### Target

Container Image

### Scanner

Vulnerability

### Output Format

JSON

### Mode

Standalone

### Operating System

Mac OS Sonoma, also in an AWS Lambda

### Version

```bash
Version: 0.52.2
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-06-24 18:11:15.08361159 +0000 UTC
  NextUpdate: 2024-06-25 00:11:15.083611369 +0000 UTC
  DownloadedAt: 2024-06-24 20:18:54.017719 +0000 UTC
```

**Comment by DimNik9** [comment]
Any updates on this?

**Comment by DmitriyLewen** [comment]
Hello @neo-travis-friesen @DimNik9 

Take a look on [open VEX](https://aquasecurity.github.io/trivy/v0.53/docs/supply-chain/vex/#openvex) (or other VEX formats)
You can ignore a specific version for each package, filter packages by image digest, write a description of why you are ignoring this vulnerability, etc.
